mitiq/shadows/quantum_processing.py

```diff
@@ -82,44 +82,51 @@ def random_pauli_measurement(
     circuit: cirq.Circuit,
     n_total_measurements: int,
     executor: Callable[[cirq.Circuit], MeasurementResult],
-) -> Tuple[NDArray[Any], NDArray[Any]]:
-    r"""
-    Given a circuit, perform random Pauli measurements on the circuit and
-    return outcomes. The outcomes are represented as a string where a
-    z-basis measurement outcome of :math:`|0\rangle` corresponds to 1, and
-    :math:`|1\rangle` corresponds to -1.
+    qubits: Optional[List[cirq.Qid]] = None,
+) -> Tuple[List[str], List[str]]:
+    r"""This function performs random Pauli measurements on a given circuit and
+    returns the outcomes. These outcomes are represented as a tuple of two
+    lists of strings.
 
     Args:
-        circuit: Cirq circuit.
-        n_total_measurements: number of snapshots.
-        executor: A callable which runs a circuit and returns a single
+        circuit: A Cirq circuit.
+        n_total_measurements: The number of snapshots.
+        executor: A callable that runs a circuit and returns a single
             bitstring.
+        qubits: The qubits to be measured. If None, all qubits in the circuit
+            will be measured.
 
     Warning:
-        The ``executor`` must return a ``MeasurementResult``
-        for a single shot (i.e. a single bitstring).
+        The `executor` must return a `MeasurementResult` for a single shot,
+        i.e., a single bitstring.
 
     Returns:
-        Tuple of two numpy arrays. The first array contains
-        measurement outcomes (-1, 1) while the second array contains the
-        indices for the sampled Pauli's ("X", "Y", "Z").
-        This implies that local Clifford rotations plus z-basis measurements
-        are effectively equivalent to random Pauli measurements.
-        Each row of the arrays corresponds to a distinct snapshot or sample,
-        while each column corresponds to measurement outcomes
-        and random Pauli measurement on a different qubit.
+        A tuple of two lists of strings, each of length equal to
+        `n_total_measurements`. Each element in the list is a string, formatted
+        as follows:
+        - First list: The computational basis of
+        circuit qubits, e.g.
+        '01...1'=:math:`|0\rangle |1\rangle...|1\rangle`.
+        - Second list: The local Pauli measurement performed on each qubit.
+        e.g. 'XY...Z' signifies a local X-basis measurement on the
+        first qubit, a local Y-basis measurement on the second qubit,
+        and a local Z-basis measurement on the last qubit in the circuit.
```

Comment:
```suggestion
        Tuple containing two lists of strings, each of length equal to
        `n_total_measurements`. Strings in the first list are sequences of 0's and
        1's, which represent qubit measurements in the computational basis (e.g.
        "01001"). Strings in the second list are sequences of Pauli operators
        performed on the circuit (e.g. "XZZII").
```

Reply:
Thanks for your suggestion Misty, Please note that the change of the doc has a minor mistake, the string in the second list just represents the Pauli-basis measurement performed on each qubit, not the "sequences of Pauli operators performed" (I think you mean rotation unitaries, but actually it's not), that's why it can't be "I". Just a minor suggestion, please Let me know before you modify it!